src/lib/inputs/InputCheckbox/InputCheckbox.tsx

```diff
@@ -30,23 +28,12 @@ export const InputCheckbox = ({
   const [checked, setChecked] = useState(!!fieldProps?.value || !!props.checked);
```

Comment:
So what we can do, I think is set a `fieldValue` variable at the top of the component scope:

```js
const fieldValue = !!fieldProps?.value || !!props.checked;
```

Then you can use that elsewhere, like here:

```js
const [checked, setChecked] = useState(fieldValue);
```

and here:

```js
useEffect(() => {
  setChecked(fieldValue);
}, [fieldValue]);
```

Then the question becomes, do you even need the `useEffect`? Or can you just use the `fieldValue`?

Reply:
I just went through this with the `InputColor` component, and I was able to simplify it quite a bit, as well.

Reply:
yeah let me try to use just fieldValue without an effect and see what happens!